### PR TITLE
ask for review

### PR DESCRIPTION
**Pull request by AnthonyTC89**

### CODE REVIEW

File: iterator_methods.rb [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    self
+  end
+
+  def my_inject
+    memo = self[0]
+    (length - 1).times do |i|
+      memo = yield memo, self[i + 1]
+    end
+    memo
+  end
```
Comment (maelfosso) [inline iterator_methods.rb:74]: Not working
```
> [5,6,7,8,9,10].inject(1) { |product, n| product * n }
 => 151200 
> [5,6,7,8,9,10].my_inject(1) { |product, n| product * n }
Traceback (most recent call last):
        3: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        2: from (irb):5
        1: from iterator_methods.rb:66:in `my_inject'
ArgumentError (wrong number of arguments (given 1, expected 0))
```

File: iterator_methods.rb [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    true
+  end
+
+  def my_count
+    count = 0
+    length.times do |i|
+      count += 1 if yield self[i]
+    end
+    count
+  end
```
Comment (maelfosso) [inline iterator_methods.rb:53]: Not working

```
> ary = [1, 2, 4, 2]
 => [1, 2, 4, 2] 
> ary.count
 => 4 
> ary.my_count
Traceback (most recent call last):
        5: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        4: from (irb):10
        3: from iterator_methods.rb:47:in `my_count'
        2: from iterator_methods.rb:47:in `times'
        1: from iterator_methods.rb:48:in `block in my_count'
LocalJumpError (no block given (yield))
> ary.count(2)
 => 2 
> ary.my_count(2)
Traceback (most recent call last):
        3: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        2: from (irb):12
        1: from iterator_methods.rb:45:in `my_count'
ArgumentError (wrong number of arguments (given 1, expected 0))
```

File: iterator_methods.rb [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    end
+    false
+  end
+
+  def my_none?
+    length.times do |i|
+      return false if yield self[i]
+    end
+    true
+  end
```
Comment (maelfosso) [inline iterator_methods.rb:43]: Some problems

```
> [nil, false, true].none? 
 => false 
> [nil, false, true].my_none? 
Traceback (most recent call last):
        5: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        4: from (irb):22
        3: from iterator_methods.rb:39:in `my_none?'
        2: from iterator_methods.rb:39:in `times'
        1: from iterator_methods.rb:40:in `block in my_none?'
LocalJumpError (no block given (yield))
```

File: iterator_methods.rb [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+  def my_any?
+    length.times do |i|
+      return true if yield self[i]
+    end
+    false
+  end
```
Comment (maelfosso) [inline iterator_methods.rb:34]: Not working

```
> [nil, true, 99].any? 
 => true 
> [nil, true, 99].my_any? 
Traceback (most recent call last):
        5: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        4: from (irb):29
        3: from iterator_methods.rb:32:in `my_any?'
        2: from iterator_methods.rb:32:in `times'
        1: from iterator_methods.rb:33:in `block in my_any?'
LocalJumpError (no block given (yield))
```

File: iterator_methods.rb [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    end
+    aux_array
+  end
+
+  def my_all?
+    length.times do |i|
+      return false unless yield self[i]
+    end
+    true
+  end
```
Comment (maelfosso) [inline iterator_methods.rb:25]: Not working

```
> [nil, true, 99].all? 
 => false 
> [nil, true, 99].my_all? 
Traceback (most recent call last):
        5: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        4: from (irb):37
        3: from iterator_methods.rb:25:in `my_all?'
        2: from iterator_methods.rb:25:in `times'
        1: from iterator_methods.rb:26:in `block in my_all?'
LocalJumpError (no block given (yield))
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello,

Thank you for this code that I had the honor to read.
Before moving on, please read the comment below and do the corresponding changes.
Don't forget to deal with `no block given`.

Thanks.

File: iterator_methods.rb (hunk 1 above)
Comment (AnthonyTC89) [reply iterator_methods.rb:74]: ok.. the method inject need arguments.. it's done

File: iterator_methods.rb (hunk 2 above)
Comment (AnthonyTC89) [reply iterator_methods.rb:53]: method my_count with arguments.. it's done

File: iterator_methods.rb (hunk 3 above)
Comment (AnthonyTC89) [reply iterator_methods.rb:43]: my_none without a block done

File: iterator_methods.rb (hunk 4 above)
Comment (AnthonyTC89) [reply iterator_methods.rb:34]: my_any without block done

File: iterator_methods.rb (hunk 5 above)
Comment (AnthonyTC89) [reply iterator_methods.rb:25]: my_all without block.. done!

Comment (maelfosso) [reply iterator_methods.rb:25]: :+1:

File: iterator_methods.rb (hunk 4 above)
Comment (maelfosso) [reply iterator_methods.rb:34]: :+1:

File: iterator_methods.rb (hunk 3 above)
Comment (maelfosso) [reply iterator_methods.rb:43]: :+1:

File: iterator_methods.rb (hunk 2 above)
Comment (maelfosso) [reply iterator_methods.rb:53]: :+1:

**Review by maelfosso — APPROVED** [review]
# Good job :+1: 

You can merge and submit the project completion form.
Good Luck with your next project.

**Pull request merged**